src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+from typing import Any, Dict, List, Tuple
+
+from django.db import transaction
+
+from sentry.issues.escalating import query_groups_past_counts
+from sentry.issues.escalating_issues_alg import generate_issue_forecast
+from sentry.models import Group, GroupStatus
+from sentry.models.groupforecast import GroupForecast
+from sentry.tasks.base import instrumented_task
+from sentry.utils.json import JSONData
+
+BATCH_SIZE = 1000
+GroupForecastTuple = Tuple[Group, List[int]]
+ParsedGroupCount = Dict[int, Dict[str, Any]]
```

Comment:
nit: I prefer `Mapping` over `Dict` if you don't actually need `ParsedGroupCount` to be mutable.

Reply:
In this case I need `ParsedGroupCount` to be mutable, but I'll keep that in mind for the future